gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -377,6 +378,13 @@ public void apply(JobListener jobListener, JobContext jobContext)
           jobState.setState(JobState.RunningState.COMMITTED);
           isWorkUnitsEmpty = true;
           return;
+        } else {
```

Comment:
how about reporting `workUnitsCreated=0` when there is no workUnits instead of reporting `WorkUnitsEmpty` event?

Reply:
This depends on the consumer of the event. In our case we consume these events in time-series and the workUnitsCreated event is set to 0 every minute irrespective of this event emitted or otherwise and hence workUnitsEmpty helps to distinguish in this scenario.
We could potentially emit both by removing the "else" clause.